watertap/core/util/initialization.py

```diff
@@ -136,3 +138,44 @@ def assert_no_degrees_of_freedom(blk):
 
     """
     check_dof(blk, True)
+
+
+def interval_initializer(blk, feasibility_tol=1e-6):
+    """
+    Improve the initialization of ``blk`` utilizing interval arithmetic.
+
+    Keyword Arguments:
+        blk : block to initialize
+        feasibility_tol : tolerance to use for FBBT (default: 1e-6)
+
+    Returns:
+        None
+
+    """
+
+    bound_cache = ComponentMap()
+
+    for v in blk.component_data_objects(Var, active=True, descend_into=True):
+        bound_cache[v] = v.bounds
+
+    fbbt(blk, feasibility_tol=feasibility_tol, deactivate_satisfied_constraints=False)
+
+    for v, bounds in bound_cache.items():
+        if v.value is None:
+            # set to the bound nearer 0
+            v.set_value(0, skip_validation=True)
+        if v.lb is not None:
+            if v.lb == v.ub:
+                v.set_value(v.lb, skip_validation=True)
+                continue
+            if v.value < v.lb:
+                # print(f"projecting {v.name} at value {v.value} onto lower bound {v.lb}")
+                v.set_value(v.lb, skip_validation=True)
+        if v.ub is not None:
+            if v.value > v.ub:
+                # print(f"projecting {v.name} at value {v.value} onto upper bound {v.ub}")
+                v.set_value(v.ub, skip_validation=True)
+
+    for v, bounds in bound_cache.items():
```

Comment:
Can this be done as part of the previous loop? I think that once you've adjusted the value there is nothing stopping you from immediately resetting the bounds.

Reply:
For reasons I do not entirely understand, the bounds are not correctly re-set if you do this in the previous loop.

Reply:
Odd - I would not have thought  there would be any difference.